process/smartContract/hooks/blockChainHook.go

```diff
@@ -485,19 +496,7 @@ func (bh *BlockChainHookImpl) GetESDTToken(address []byte, tokenID []byte, nonce
 	}
 
 	esdtTokenKey := []byte(core.ElrondProtectedKeyPrefix + core.ESDTKeyIdentifier + string(tokenID))
-	if nonce > 0 {
-		esdtTokenKey = append(esdtTokenKey, big.NewInt(0).SetUint64(nonce).Bytes()...)
-	}
-
-	value, err := userAcc.AccountDataHandler().RetrieveValue(esdtTokenKey)
-	if err != nil {
-		return nil, err
-	}
-	if len(value) == 0 {
-		return esdtData, nil
-	}
-
-	err = bh.marshalizer.Unmarshal(esdtData, value)
+	esdtData, _, err = bh.nftStorageHandler.GetESDTNFTTokenOnDestination(userAcc, esdtTokenKey, nonce)
```

Comment:
is this backwards compatible? if calling `GetESDTToken` before the enable epoch

Reply:
yes. epoch enable logic is inside the new nftStorageHandler